rust/lance-core/src/utils/tracing.rs

```diff
@@ -84,3 +84,4 @@ pub const DATASET_DROPPING_COLUMN_EVENT: &str = "dropping_column";
 pub const DATASET_DELETING_EVENT: &str = "deleting";
 pub const DATASET_COMPACTING_EVENT: &str = "compacting";
 pub const DATASET_CLEANING_EVENT: &str = "cleaning";
+pub const DATASET_LOADING_EVENT: &str = "loading";
```

Comment:
Could you describe the different between opening and loading? I think that would be confusing to most people.

Reply:
yeah I think we should only have one. If we want to differentiate between cached open vs non-cached open, Weston has a good point at https://github.com/lancedb/lance/discussions/4234#discussioncomment-13760219 that we should trace that through file opening trace events.

Reply:
> Could you describe the different between opening and loading?

It's misunderstood. I thought we had a clear path to distinguish `write` and `read`.

* `writing` for only writing data;
* `opening` for only opening and reading the existing dataset;

The reality is:

* currently, read and write are both called `load` method;
* The current `opening` could not track the main scenario(`lance.dataset('xxxx')`)

I did not remove the `opening` event. Because I'm not sure if the `opening` is still useful for someone who called the `open` method in Rust.